src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs

```diff
@@ -58,9 +58,6 @@ namespace System.Threading.Tasks
     [DebuggerDisplay("Id = {Id}, Status = {Status}, Method = {DebuggerDisplayMethodDescription}, Result = {DebuggerDisplayResultDescription}")]
     public class Task<TResult> : Task
     {
-        /// <summary>A cached task for default(TResult).</summary>
```

Comment:
This generates before-field-init constructor. The constructor should be running lazily only when the field is accessed.

Reply:
Is this working around Mono running before-field-init constructors earlier than required? It would be better to fix that in Mono for all before-field-init constructors.

Reply:
> This generates before-field-init constructor. The constructor should be running lazily only when the field is accessed. I do not see how this helps.

It's making the AOT compiler compile a cctor for every instance of Task<T>, even if it is never used

Reply:
> Is this working around Mono running before-field-init constructors earlier than required? It would be better to fix that in Mono for all before-field-init constructors.

That is a good point, after we make that fix, this won't show up in startup traces anymore.

I still feel this might be a trimming/AOT win. But I guess it depends on how prevalent `Task.FromResult` is at different instantiations.

I'll close this tomorrow if I don't see an AOT impact